projectmanagement/src/main/java/life/qbic/projectmanagement/application/ProjectInformationService.java

```diff
@@ -70,61 +61,61 @@ public List<ProjectPreview> queryPreview(String filter, int offset, int limit,
   @PostAuthorize("hasPermission(returnObject,'VIEW_PROJECT')")
   public Optional<Project> find(String projectId) {
     Objects.requireNonNull(projectId);
-    log.debug("Search for project with id: " + projectId);
-    return projectRepository.find(ProjectId.parse(projectId));
+    return Optional.ofNullable(loadProject(projectId));
   }
 
   @PostAuthorize("hasPermission(returnObject,'VIEW_PROJECT')")
-  private Project loadProject(ProjectId projectId) {
+  private Project loadProject(String projectId) {
     Objects.requireNonNull(projectId);
-    return projectRepository.find(projectId).orElseThrow(() -> new ProjectManagementException(
-            "Project with id" + projectId.toString() + "does not exit anymore")
+    ProjectId parsedProjectId;
```

Comment:
what happened here?

Reply:
The internal method now also checks if the provided projectId is parseable and throws an exception is not possible.

Reply:
I do not think that you need to change that here. You can leave the method signature with the ProjectId class which ensures the correct formatting of the project identifier.

Reply:
However this leads to the previous situation in which every method that employs the loadProject() method needs to parse the ProjectID seperately beforehand.